src/fdc/DiskImageUtils.cc

```diff
@@ -189,34 +192,42 @@ static SetBootSectorResult setBootSector(
 	unsigned nbRootDirSectors = nbDirEntry / 16;
 	unsigned rootDirStart = nbReservedSectors + nbFats * nbSectorsPerFat;
 	unsigned firstDataSector = rootDirStart + nbRootDirSectors;
-	return {nbReservedSectors, firstDataSector, descriptor};
+	return {nbFats, nbSectorsPerFat, nbReservedSectors, rootDirStart, firstDataSector, descriptor};
 }
```

Comment:
I don't really like this. It's fine for 2 or 3 return values. But for 6 values it becomes too easy to accidentally enumerate them in the wrong order. And the compiler can't check that indeed `nbFats` corresponds to `fatCount` etc.    ... Euhm, and I think the order is indeed already wrong :-)     You swapped the first two.

I'd suggest to define a variable `SetBootSectorResult result` inside `setBootSector()`, fill in the fields, and return that variable.

Reply:
Ah sorry I meant to do that, but rushed the commit and forgot that final step. Funny how it immediately goes wrong, too :).